ouroboros-consensus/src/Ouroboros/Consensus/Util/MonadSTM/NormalForm.hs

```diff
@@ -14,14 +15,21 @@ module Ouroboros.Consensus.Util.MonadSTM.NormalForm (
 import           GHC.Stack
 import           NoThunks.Class (NoThunks (..), unsafeNoThunks)
 
-import           Control.Monad.Class.MonadSTM.Strict hiding (newEmptyTMVarIO,
-                     newTMVar, newTMVarIO, newTVar, newTVarIO,
-                     newTVarWithInvariantIO)
+
 import           Ouroboros.Consensus.Util.MonadSTM.StrictMVar hiding
                      (newEmptyMVar, newEmptyMVarWithInvariant, newMVar,
                      newMVarWithInvariant)
 
-import qualified Control.Monad.Class.MonadSTM.Strict as Strict
+import           Control.Concurrent.Class.MonadSTM.Strict.TMVar as StrictSTM hiding
+                     (newTMVar, newTMVarIO)
+import           Control.Concurrent.Class.MonadSTM.Strict.TVar as StrictSTM hiding
+                     (newTVar, newTVarIO, newTVarWithInvariantIO)
+-- TODO: use strict versions of 'TQueue' and 'TBQueue'
```

Comment:
Could we give a rationale on why this is required? Eg what would happen if we don't use the strict versions of these queue types?

Reply:
I updated the todo.